### PR TITLE
[enhancement] Refactor AlterCluse add isNewAlterTableClause function

### PR DESCRIPTION
**Pull request by zhuxt2015**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 2 / 5 (40.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/DropIndexClause.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/TableRenameClause.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/CreateIndexClause.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/AlterTableStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AlterClause.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/analysis/AlterClause.java [hunk 1]
```diff
@@ -42,4 +42,8 @@ public Map<String, String> getProperties() {
     public AlterOpType getOpType() {
         return opType;
     }
+
+    public boolean isNewAlterTableClause() {
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/analysis/AlterClause.java:46]: I think `isSupportNewPlanner` is better

Comment (zhuxt2015) [reply fe/fe-core/src/main/java/com/starrocks/analysis/AlterClause.java]: yes, same to StatementBase

**Comment by zhuxt2015** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 2 / 5 (40.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/DropIndexClause.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/TableRenameClause.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/CreateIndexClause.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/AlterTableStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AlterClause.java|1|1|100.00%|

**Review by kangkaisen — APPROVED** [review]

**Comment by padmejin** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 2 / 5 (40.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/DropIndexClause.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/TableRenameClause.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/CreateIndexClause.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/AlterTableStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AlterClause.java|1|1|100.00%|

**Review by imay — APPROVED** [review]

**Pull request merged**